app/commands/v2/put_link_set.rb

```diff
@@ -9,7 +9,21 @@ def call
           version.increment
           version.save! if link_set.valid?
 
-          link_set.links = merge_links(link_set.links, link_params.fetch(:links))
+          links_hash = link_params.fetch(:links)
+
+          links_hash.each do |link_type, target_content_ids|
+            if target_content_ids.empty?
+              link_set.links.where(link_type: link_type).delete_all
+            else
+              old_target_ids = link_set.links.where(link_type: link_type).pluck(:target_content_id)
+
+              ids_to_be_deleted = old_target_ids - target_content_ids
+              link_set.links.where(link_type: link_type, target_content_id: ids_to_be_deleted).delete_all
+
+              ids_to_be_created = target_content_ids - old_target_ids
+              link_set.links.where(link_type: link_type, target_content_id: ids_to_be_created).create
+            end
+          end
```

Comment:
I wonder if we could move this diff'ing logic to the model layer.

I think you could probably make `#links` a virtual attribute on the model that this method sets. The model then performs the diff itself and creates/deletes records. I'm not sure exactly how this would look.